### PR TITLE
Fix #2981, which causes union relationship fields to be represented as a List even when defined as a single element.

### LINKED ISSUE #2981: Unable to create a nested object when type is an union
**Describe the bug**
When I try to create a nested object I get a null exception.

**Type definitions**
```graphql
type Book {
  originalTitle: String!
  translatedTitle: BookTitle @relationship(type: "TRANSLATED_BOOK_TITLE", direction: IN) 
  isbn: String!
}

union BookTitle = BookTitle_SV | BookTitle_EN

type BookTitle_SV {
  book: Book! @relationship(type: "TRANSLATED_BOOK_TITLE", direction: OUT)
  value: String!
}

type BookTitle_EN {
  book: Book! @relationship(type: "TRANSLATED_BOOK_TITLE", direction: OUT)
  value: String!
}
```

**To Reproduce**

1. Run a server with the code
2. Execute the following Mutation
```graphql
mutation CreateBooks {
  createBooks(
    input: [
      {
        isbn: "123"
        originalTitle: "Original title"
        translatedTitle: {
          BookTitle_SV: {
            create: { node: { value: "Exempel på svensk titel" } }
          }
        }
      }
    ]
  ) {
    info {
      bookmark
      nodesCreated
      relationshipsCreated
    }
    books {
      isbn
      originalTitle
      translatedTitle {
        ... on BookTitle_SV {
          value
        }
        ... on BookTitle_EN {
          value
        }
      }
    }
  }
}
```
4. Then run the following Mutation
```graphql
mutation UpdateBooks {
  updateBooks(
    where: { isbn: "123" }
    create: { translatedTitle: { BookTitle_EN: [{ node: { value: "English book title" } }] } }
  ) {
    info {
      nodesCreated
      nodesDeleted
      relationshipsCreated
      relationshipsDeleted
    }
    books {
      isbn
      originalTitle
      translatedTitle {
        ... on BookTitle_SV {
          value
        }
        ... on BookTitle_EN {
          value
        }
      }
    }
  }
}
```
6. See error
```
TypeError: Cannot convert undefined or null to object",
          "    at Function.entries (<anonymous>)",
          "    at createCreateAndParams (/Users/x/y/node_modules/@neo4j/graphql/dist/translate/create-create-and-params.js:238:44)",
```
**Expected behavior**
I looked at the code. It seems that in the file `translate-update.js` the snippet below expect `create.node` to be a nested object when it in fact an array with objects. If I type `create[0].node` in watch expressions I get the correct values:
```javascript
 const createAndParams = (0, create_create_and_params_1.default)({
                        context,
                        callbackBucket,
                        node: refNode,
                        input: create.node,
                        varName: nodeName,
                        withVars: [...withVars, nodeName],
                        includeRelationshipValidation: false,
                    });

```
**Screenshots**
N/A

**System (please complete the following information):**
N/A

**Additional context**
Not sure if this has been registered, did try to do a search but didn't find anything.

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Fix 2981

Bug 2981 is caused when the user defines a 1 to 1 relationship with the nested node be a union type. The union fields of the `CreateFieldInput` were defined as a List without checking the cardinality defined by the user.

## Complexity

Complexity: Very Low

# Issue

Closes #2981
Blocked by: #3027

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-3029.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 44 | 56 | 1134 |
|  aggregations.NestedAggregation | 14551 | 14551 | 121 | 145 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 107 | 135 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3843 | 3954 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 3473 | 3437 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 63 | 96 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 60 | 108 | 2174 |
|  aggregations.DeeplyNestedCount | 13070329 | 13070333 | 5619 | 5935 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 193 | 177 | 600 |
|  connect.createAndConnect | 14424 | 14424 | 233 | 279 | 3003 |
|  connections.Connection | 14082 | 14082 | 104 | 96 | 2174 |
|  connections.NestedConnection | 41466 | 41468 | 179 | 163 | 4516 |
|  connections.ConnectionWithSort | 2282 | 2282 | 76 | 82 | 1040 |
|  connections.ConnectionWithSortAndCypher | 2282 | 2282 | 224 | 350 | 1040 |
|  create.SimpleMutation | 7 | 7 | 68 | 77 | 1 |
|  cypher-directive.TopLevelCypherDirective | 116092 | 116092 | 895 | 1008 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 147609 | 147609 | 349 | 428 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 443 | 364 | 1040 |
|  delete.NestedDeleteInUpdate | 22544 | 22544 | 158 | 226 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 88 | 54 | 1040 |
|  query.SimpleQuery | 15160 | 15160 | 61 | 101 | 2174 |
|  query.QueryWhere | 9709 | 9709 | 50 | 47 | 2166 |
|  query.SimpleQueryWithNestedWhere | 9887 | 9887 | 67 | 69 | 2166 |
|  query.Nested | 10092037 | 10092037 | 11422 | 11853 | 2008534 |
|  query.NestedWithFilter | 40401 | 40401 | 235 | 131 | 4511 |
|  query.OrFilterOnRelationships | 43613 | 43778 | 253 | 217 | 2170 |
|  query.OrFilterOnRelationshipsAndNested | 37938 | 37046 | 231 | 454 | 2170 |
|  query.QueryWithNestedIn | 13213 | 13381 | 68 | 88 | 1156 |
|  query.NestedConnectionWhere | 9834 | 9834 | 76 | 74 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10049 | 10049 | 100 | 112 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6343 | 6343 | 158 | 232 | 1134 |
|  query.NestedWithRelationshipSingleFilters | 3024400 | 3024400 | 679 | 709 | 1003942 |
|  query.Fulltext | 96 | 96 | 32 | 35 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 52 | 58 | 84 |
|  sorting.SortMultipleTypes | 2513 | 2513 | 79 | 86 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1459 | 1459 | 216 | 428 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 100 | 174 | 2174 |
|  sorting.SortDeeplyNestedFields | 42196 | 42196 | 205 | 221 | 4516 |
|  sorting.SortWithTopLevelCypher | 2119 | 2119 | 141 | 208 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 75 | 96 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 77 | 95 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 359 | 406 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 355 | 443 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 82 | 95 | 1040 |

</details>
        
Old Schema Generation: 39.716s
Schema Generation: 40.635s
Old Subgraph Schema Generation: 59.361s
Subgraph Schema Generation: 1:00.879 (m:ss.mmm)

**Comment by darrellwarde** [comment]
With squash and merge now being the preference, we need more descriptive pull request titles! 😉 Review incoming.

File: .changeset/slow-spoons-joke.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@neo4j/graphql": patch
+---
+
+Fixed a bug that generates a relationship field of a union type as a List, in some inputs, even if defined as a 1 to 1 relationship field by the user.
```
Comment (darrellwarde) [inline .changeset/slow-spoons-joke.md:5]: Great changelog entry! 👏

**Review by darrellwarde — APPROVED** [review]
LGTM with the PR title changed! 😄

**Review by angrykoala — APPROVED** [review]

**Pull request merged**